packages/components/src/unit-control/stories/index.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { UnitControl } from '../';
+import { CSS_UNITS } from '../utils';
+
+const meta: ComponentMeta< typeof UnitControl > = {
+	component: UnitControl,
+	title: 'Components (Experimental)/UnitControl',
+	argTypes: {
+		size: {
+			control: { type: 'select' },
+		},
+		onChange: {
+			control: { type: null },
+		},
+		onUnitChange: {
+			control: { type: null },
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+			exclude: [ '__unstableInputWidth', '__unstableStateReducer' ],
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const DefaultTemplate: ComponentStory< typeof UnitControl > = ( {
+	onChange,
+	...args
+} ) => {
+	const [ value, setValue ] = useState< string | undefined >( '10px' );
+
+	return (
+		<div style={ { maxWidth: '100px' } }>
+			<UnitControl
+				{ ...args }
+				value={ value }
+				onChange={ ( v, extra ) => {
```

Comment:
The fact that we have to set the `onChange` callback _within_ the template causes Storybook to display an empty callback in the code snippet, wondering if there's a way to avoid this?

![Screenshot 2022-03-09 at 18 51 25](https://user-images.githubusercontent.com/1083581/157501356-272901a9-9c91-4bc4-8e89-99c4a3564379.png)

Reply:
Oh, interesting... and we can't pass it in directly because we still need to call `setValue` within this component, is that right? For what it is rendering, in this case is it rendering the `() => {}` because in the `UnitControl` component it defaults `onChange` to a `noop` function? It's probably not _terrible_ for the code snippet to not be showing a useful callback, but I see what you mean — it'd be better if it had a more real-world example usage 🤔

Reply:
I went ahead and removed the default `noop` fallback (both in the code, and as a `@default` value in the Docs annotations), but that doesn't seem to solve the issue.

It looks like it's showing `{() => {}}` as if it wasn't able to "introspect" more into the story's code, and actually show the real code for the `onChange` callback.

I may need to look more into this

Reply:
Going to look more into it, will report back!

Reply:
I spent some time looking into why `onChange` is displayed as `() => {}`.

- The proper way to fix it would be to change the source type from `dynamic` (current behaviour) to `code`, as explained in details in the [docs](https://github.com/storybookjs/storybook/blob/next/docs/writing-docs/doc-block-source.md)
- I tried changing this config option in `UnitControl`'s story, and the code snippet stopped being displayed completely. After some research, it seems like a known bug (https://github.com/storybookjs/storybook/issues/12561) without a fix yet.

I guess we'll have to keep the source as-is, until a fix is released.

> Speaking of event handlers, I was going to ask if the [Actions addon](https://storybook.js.org/docs/react/essentials/actions) would be useful or not.

I went ahead and [added the addon to the project](https://github.com/WordPress/gutenberg/pull/39320/commits/47537dbae4aa1ff584b55ddfc852a7bc82b3b64f). I then experimented with it, and decided to only [enable it for the `onChange` callback](https://github.com/WordPress/gutenberg/pull/39320/commits/af9553f4128fc5abf19676e46600cfc8751ab70e), since enabling it on the `onUnitChange` callback was generating a weird output in the code snippet (`noRefCheck()`):

![Screenshot 2022-03-10 at 19 01 45](https://user-images.githubusercontent.com/1083581/157731806-f0d9f170-1756-49ad-a71d-9392c92e210c.png)

Reply:
Quick update — it looks like the Storybook bug that I had linked previously has been closed and [marked as fixed in the release 7](https://github.com/storybookjs/storybook/issues/12561#issuecomment-1416710586), with the `@storybook/csf-plugin` package released.

Let's see if we need to take action at all, or if updating to Storybook 7 will be enough 🤞